pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+package spec
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/go-openapi/jsonreference"
+	openapi_v2 "github.com/google/gnostic/openapiv2"
+)
+
+// Interfaces
+type GnosticCommonValidations interface {
+	GetMaximum() float64
+	GetExclusiveMaximum() bool
+	GetMinimum() float64
+	GetExclusiveMinimum() bool
+	GetMaxLength() int64
+	GetMinLength() int64
+	GetPattern() string
+	GetMaxItems() int64
+	GetMinItems() int64
+	GetUniqueItems() bool
+	GetMultipleOf() float64
+	GetEnum() []*openapi_v2.Any
+}
+
+func (k *CommonValidations) FromGnostic(g GnosticCommonValidations) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	max := g.GetMaximum()
+	if max != 0 {
+		k.Maximum = &max
+	}
+
+	k.ExclusiveMaximum = g.GetExclusiveMaximum()
+
+	min := g.GetMinimum()
+	if min != 0 {
+		k.Minimum = &min
+	}
+
+	k.ExclusiveMinimum = g.GetExclusiveMinimum()
+
+	maxLen := g.GetMaxLength()
+	if maxLen != 0 {
+		k.MaxLength = &maxLen
+	}
+
+	minLen := g.GetMinLength()
+	if minLen != 0 {
+		k.MinLength = &minLen
+	}
+
+	k.Pattern = g.GetPattern()
+
+	maxItems := g.GetMaxItems()
+	if maxItems != 0 {
+		k.MaxItems = &maxItems
+	}
+
+	minItems := g.GetMinItems()
+	if minItems != 0 {
+		k.MinItems = &minItems
+	}
+
+	k.UniqueItems = g.GetUniqueItems()
+
+	multOf := g.GetMultipleOf()
+	if multOf != 0 {
+		k.MultipleOf = &multOf
+	}
+
+	enums := g.GetEnum()
+
+	if enums != nil {
+		k.Enum = make([]interface{}, len(enums))
+		for i, v := range enums {
```

Comment:
still a fair number of places we're ranging over pointers that could be nil... I guess we should always check nil and `continue` if nil, rather than accessing fields or calling methods on those things.